jena-core/src/main/java/org/apache/jena/graph/impl/TransactionHandlerBase.java

```diff
@@ -69,5 +81,17 @@ public <T> T calculate( Supplier<T> action ) {
         catch (JenaException e) { abort(e) ; throw e ; }
         catch (Throwable e)     { abort(e) ; throw new JenaException(e) ; }
     }
+
+    /**
+     * Calculate inside a transaction if transactions supported - calculate anyway if transactions not supported.
+     */
+    @Override
+    public <T> T calculateAlways( Supplier<T> action ) {
+        if ( ! transactionsSupported() )
+            return calculate(action);
+        else
+            return action.get();
```

Comment:
@afs just the normal OCD, but in `executeAlways` the logic appears to be a little different?

- `executeAlways`:
  - no transactions? call the action method
  - transactions? call another method
- `calculateAlways`:
  - no transactions? call another method
  - transactions? call the action method

Also bugged me that in one we control the flow with `if` + `return`, but here we just use `if` and `else` :smile: I know right? Sorry!

Reply:
Good catch. The logic is reversed in `calculate`.

Not that there is much use of `calculate` in the code, but it is in the test suite. That also needs checking.